src/lib/features/feature-lifecycle/feature-lifecycle-read-model.ts

```diff
@@ -17,6 +21,61 @@ export class FeatureLifecycleReadModel implements IFeatureLifecycleReadModel {
         this.db = db;
     }
 
+    async getStageCount(): Promise<StageCount[]> {
+        const { rows } = await this.db.raw(`
+            SELECT
+                stage,
+                COUNT(*) AS feature_count
+            FROM (
+                SELECT DISTINCT ON (feature)
+                    feature,
+                    stage,
+                    created_at
+                FROM
+                    feature_lifecycles
+                ORDER BY
+                    feature, created_at DESC
```

Comment:
What role does ordering play here? Does it improve performance in any way?

Reply:
Disregard, reading the rest of the query again I see you're selecting distinct, so you want the most recent status!